### ISSUE TITLE
PageSelectorType returns Block instead of Page entity

### ISSUE THREAD

**Issue by jorenvh1** [sonata-project/SonataPageBundle#989]
I'm trying to add a PageSelectorType form field to my block form so I can choose a page for the current site.

I'm doing it like this:
```$formMapper
            ->add(
                'settings',
                'sonata_type_immutable_array',
                array(
                    'keys' => array(
                        array(
                            'banner_page',
                            PageSelectorType::class,
                            array(
                                'model_manager' => $formMapper->getAdmin()->getModelManager(),
                                'class' => $formMapper->getAdmin()->getClass(),
                                'site' => $block->getPage()->getSite(),
                                'required' => true,
                                'label' => 'form.label_page',
                            ),
                        ),
                        array($imageBuilder, null, array()),
                        array('title', TextType::class, array('required' => false)),
                        array('cta_button_text', TextType::class, array('required' => false)),
                    ),
                    'translation_domain' => 'SonataPageBundle',
                )
            ); 
```

This renders a correct page selector but when I'm saving the block and I put a breakpoint inside the prepersist or preupdate function I see that the banner_page property inside my block settings is a Block entity instead of a Page entity. This means that the chosen page id in the form gets parsed to a block entity with the same id of the chosen page. This also causes the form to do weird stuff when selecting a page with an id if there's not block with that same id...

tl;dr => pageselector translates page id to a Block entity instead of a Page entity...

**Comment by jorenvh1** [comment]
I feel like this has something to do
